internal/proxy/proxy.go

```diff
@@ -61,18 +64,28 @@ type Config struct {
 	Dialer cloudsql.Dialer
 }
 
-// NewConfig initializes a Config struct using the default database engine
-// ports.
-func NewConfig() *Config {
-	return &Config{}
+// Logger defines a minimal interface for logging informational or error
+// messages.
+type Logger interface {
+	// Log reports an informational message.
+	Log(msg string, data ...interface{})
+	// Error reports an error message.
+	Error(msg string, data ...interface{})
 }
 
-func (c *Config) DialerOpts() []cloudsqlconn.Option {
+func (c *Config) DialerOpts(l Logger) []cloudsqlconn.Option {
 	var opts []cloudsqlconn.Option
-	if c.Token != "" {
+	switch {
+	case c.Token != "":
+		l.Log("Using provided token for authentication\n")
```

Comment:
nit: Can we use the pattern 

- "Authorizing with the `-token` flag"
- "Authorizing with the credential file at %q"

we should also probably log a "Authorizing using the Application Default credentials" if nothing is passed.

Reply:
Nice. Done.